internal/db/schema/migrations/oss/postgres/48/01_static_kv_creds.up.sql

```diff
@@ -0,0 +1,81 @@
+begin;
+
+  -- drop constraint so we can add kv
+  alter table credential_type_enm
+    drop constraint only_predefined_credential_types_allowed;
+
+  -- Add new constraint that only allows known types
+  -- This replaces the constraint defined in 39/01_static_ssh_private_key_creds.up.sql
+  alter table credential_type_enm
+    add constraint only_predefined_credential_types_allowed
+      check (
+        name in (
+          'unspecified',
+          'username_password',
+          'ssh_private_key',
+          'kv'
+        )
+      );
+
+  create table credential_static_kv_credential (
+    public_id wt_public_id primary key,
+    store_id wt_public_id not null
+      constraint credential_static_store_fkey
+        references credential_static_store (public_id)
+        on delete cascade
+        on update cascade,
+    project_id wt_public_id not null,
+    name wt_name,
+    description wt_description,
+    create_time wt_timestamp,
+    update_time wt_timestamp,
+    version wt_version,
+
+    secrets_encrypted bytea not null
+      constraint secrets_encrypted_must_not_be_empty
+        check(length(secrets_encrypted) > 0),
+    secrets_hmac bytea not null
+      constraint secrets_hmac_must_not_be_empty
+        check(length(secrets_hmac) > 0),
+    key_id text not null
```

Comment:
This type should probably be `kms_private_id` as that is the type of the referenced column: https://github.com/hashicorp/boundary/blob/d7c4c648ec89c98d05cbbd6fee3603f9b78fab67/internal/db/schema/migrations/oss/postgres/30/04_kms_keys.up.sql#L83